components/messaging/CurrentConversation.tsx

```diff
@@ -86,6 +108,20 @@ const CurrentConversation: React.FC = () => {
 
   return (
     <div className='conversation-height mb-10 flex flex-1 flex-col justify-between bg-[#fafaf9] shadow-inner'>
+      <div className='flex flex-row p-5'>
+        <p data-testid='item-donor'>
+          <b>From: </b>
+          {itemDonor && itemDonor.username}
+        </p>
+        {itemDonor && (
+          <img
+            className='ml-2'
+            alt='user logo'
+            width='25'
+            src={itemDonor.avatar}
+          ></img>
+        )}
+      </div>
```

Comment:
I think there may be an argument for actually wrapping this here in a new component. When a donor looks at their ConversationsList they may have a number of chats related to the same item so we may need to add information on the other person to the ConversationCard.

If we separate out this item to a component named `ConversationPartner.tsx` or something we can re-use it for that case. We can then also do the evaluation on itemDonor only once

```js
itemDonor && <ConversationPartner partner={itemDonorData} />
```

Reply:
Hmm, it's not really acting as a conversation partner. It's displaying the item donor. 

This has actually made me realise there's a flaw here. When a buyer/ borrower messages the item donor, the item donor sees their own name.
<img width="932" alt="Screenshot 2024-03-12 at 15 17 27" src="https://github.com/enBloc-org/kindly/assets/114364165/31832681-34a7-4760-a6b0-4fe68da184c6">

We might need to rethink this...

Reply:
*Update* This now works completely differently. There is no need for the database call to the items table as we can pass the information from getMessagesForCurrentConversation to getProfiles that gets the username and avatar of the sender. I have extracted it into a separate component like you asked although I don't fully understand your plans for this!

Reply:
Thanks Beth.

I think this change has introduced a bug though - I have conversations where this new element isn't displayed at all.

What I think is happening is that on conversations where the other user has not responded yet you are pulling only data on the current user (only I sent a message, so all messages have my ID) and so there is no conversationPartner information to display 🙃

Reply:
> That is the expected behaviour. The conversationPartner doesn't exist unless the other person has sent a message. `From: ` suggests that we are displaying _who_ the other messages are from, it would be weird for it to be there if the other person hadn't sent a message?

Surely not - we're not displaying a user in each message, we're display one user at the top of the conversation because it can only be between two persons and the current user needs to know who they are talking to. Whether or not there have been messages traded already the conversationPartner will be the same.

Imagine the user has started a new conversation related to an item they want, but now that they've opened the chat they have no idea what the donor's name is. They would have to go all the way back to the item page to find out - instead, we display the person's name at the top of the page.

Similarly, if a Donor has several conversations going on about the same item, they need to know which person they are talking to on each separate tab. That's why we give them that information (which we need to also display on the ConversationCard).

Reply:
Okay, we can add this in but I think in that particular case (before the other person responds or sends any message) we need to change ```From: ``` to something else, maybe we just omit it and show the donor username and avatar?

To your last point, the donor cannot start a conversation so there would never be a case where they have conversations and can't see the ```From: username```